### PR TITLE
Sync assert fix

### PR DESCRIPTION
**Pull request by czue**
finally think i figured this out. was much sillier than I expected. wait for tests

@snopoke @dannyroberts

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py [hunk 1]
```diff
@@ -62,7 +62,8 @@ def get_payload(self):
                 self.restore_state.domain, list(other_ids_to_check), self.restore_state.last_sync_log.date
             ))
 
-        all_syncing = copy(case_ids_to_sync)
+        all_maybe_syncing = copy(case_ids_to_sync)
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py]: renamed for clarity

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py [hunk 2]
```diff
@@ -106,8 +108,8 @@ def get_payload(self):
         )
         self.restore_state.current_sync_log.log_format = LOG_FORMAT_SIMPLIFIED
         index_tree = IndexTree(indices=all_indices)
-        case_ids_on_phone = all_syncing
-        primary_cases_syncing = all_syncing - all_dependencies_syncing
+        case_ids_on_phone = all_synced
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py]: this is the fix - don't use everything that was checked, use everything that actually synced as the base set

**Comment by dannyroberts** [comment]
Awesome! Glad you found it so quickly!

**Comment by dannyroberts** [comment]
good to merge once tests pass

**Comment by czue** [comment]
:white_check_mark:

**Comment by snopoke** [comment]
:+1:

**Pull request merged**